utils/sl/site_tools/intelc.py

```diff
@@ -0,0 +1,657 @@
+            top = None
+            for d in dirs:
+                if os.path.exists(os.path.join(d % version, "bin", "icc")):
+                    top = d % version
+                    break
+            return top
+
+        def find_in_2010style_dir(version):
+            dirs = ('/opt/intel/Compiler/%s/*' % version)
+            # typically /opt/intel/Compiler/11.1/064 (then bin/intel64/icc)
+            dirs = glob.glob(dirs)
+            # find highest sub-version number by reverse sorting and picking first existing one.
+            dirs.sort()
+            dirs.reverse()
+            top = None
+            for d in dirs:
+                if (os.path.exists(os.path.join(d, "bin", "ia32", "icc")) or
+                        os.path.exists(os.path.join(d, "bin", "intel64", "icc"))):
+                    top = d
+                    break
+            return top
+
+        def find_in_2011style_dir(version):
+            # The 2011 (compiler v12) dirs are inconsistent, so just redo the search from
+            # get_all_compiler_versions and look for a match (search the newest form first)
+            top = None
+            for d in glob.glob('/opt/intel/composer_xe_*'):
+                # Typical dir here is /opt/intel/composer_xe_2011_sp1.11.344
+                # The _sp1 is useless, the installers are named 2011.9.x, 2011.10.x, 2011.11.x
+                m = re.search(r'([0-9]{0,4})(?:_sp\d*)?\.([0-9][0-9.]*)$', d)
+                if m:
+                    cur_ver = "%s.%s" % (m.group(1), m.group(2))
+                    if cur_ver == version and \
+                            (os.path.exists(os.path.join(d, "bin", "ia32", "icc")) or
+                             os.path.exists(os.path.join(d, "bin", "intel64", "icc"))):
+                        top = d
+                        break
+            if not top:
+                for d in glob.glob('/opt/intel/composerxe-*'):
+                    # Typical dir here is /opt/intel/composerxe-2011.4.184
+                    m = re.search(r'([0-9][0-9.]*)$', d)
+                    if m and m.group(1) == version and \
+                            (os.path.exists(os.path.join(d, "bin", "ia32", "icc")) or
+                             os.path.exists(os.path.join(d, "bin", "intel64", "icc"))):
+                        top = d
+                        break
+            return top
+
+        def find_in_2016style_dir(version):
+            # The 2016 (compiler v16) dirs are inconsistent from previous.
+            top = None
+            for d in glob.glob('/opt/intel/compilers_and_libraries_%s/linux' % version):
+                if os.path.exists(os.path.join(d, "bin", "ia32", "icc")) or os.path.exists(
+                        os.path.join(d, "bin", "intel64", "icc")):
+                    top = d
+                    break
+            return top
+
+        def find_in_intel_one_api_dir(version):
+            top = None
+            intel_one_api_path = os.path.expanduser("~/intel/oneapi")
```

Comment:
I think if you use user installed version, you should at least add /opt/intel/oneapi as the default place to look

Reply:
Ok, I'll change the default path. Is adding environment variables to modify where Scons look for icc, a "scons" thing to do?